src/BingChat/BingChatConversation.cs

```diff
@@ -169,16 +167,54 @@ public async Task<string> AskAsync(string message, CancellationToken ct = defaul
         return messages.Count > 0 ? string.Join("\n\n", messages) : null;
     }
 
-    private static async Task<HubConnection> Connect(CancellationToken ct)
+    private HttpConnectionFactory CreateConnectionFactory() => new(Options.Create(
+                new HttpConnectionOptions
+                {
+                    DefaultTransferFormat = TransferFormat.Text,
+                    SkipNegotiation = true,
+                    Transports = HttpTransportType.WebSockets,
+                    Cookies = _cookies,
+                    Headers =
+                    {
+                        ["User-Agent"] = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36 Edg/117.0.2045.47",
+                        ["Origin"] = "https://www.bing.com"
+                    }
+                }),
+            NullLoggerFactory.Instance);
+
+    private async Task<HubConnection> Connect(CancellationToken ct = default)
     {
+        var uri = this._encryptedConversationSignature == null
+            ? "wss://sydney.bing.com/sydney/ChatHub"
+            : $"wss://sydney.bing.com/sydney/ChatHub?sec_access_token={HttpUtility.UrlEncode(_encryptedConversationSignature)}";
+
         var conn = new HubConnection(
-            ConnectionFactory,
+            CreateConnectionFactory(),
             HubProtocol,
-            HubEndpoint,
+            new UriEndPoint(new Uri(uri)),
             new ServiceCollection().BuildServiceProvider(),
-            NullLoggerFactory.Instance);
+            NullLoggerFactory.Instance,
+            DefaultRetryPolicy);
 
-        await conn.StartAsync(ct);
+        //Retry on SSL connection error.
+        int retry = 0;
+        while (true)
+        {
+            try
+            {
+                await conn.StartAsync(ct);
+                break;
+            }
+            catch (WebSocketException)
+            {
+                Thread.Sleep(100);
```

Comment:
You must never use `Thread.Sleep(...)` in `async` methods or methods that can be called from async. The correct way to do it is to use `Task.Delay(...)`.